repos/system_upgrade/el7toel8/actors/checkfstaboptions/tests/test_checkfstaboptions.py

```diff
@@ -0,0 +1,87 @@
+from leapp.models import FstabEntry, StorageInfo
+from leapp.snactor.fixture import current_actor_context
+from leapp.reporting import Report
+
+
+MOUNTS_CORRECT_1 = [
+    FstabEntry(
+        fs_spec='/dev/mapper/rhel_ibm--p8--kvm--03--guest--02-root',
+        fs_file='/',
+        fs_vfstype='xfs',
+        fs_mntops='defaults',
+        fs_freq='0',
+        fs_passno='0'
+    ),
+    FstabEntry(
+        fs_spec='UUID=c3ae2d1a-8088-45c2-9482-4aacb2c075cf',
+        fs_file='/boot',
+        fs_vfstype='xfs',
+        fs_mntops='defaults',
+        fs_freq='0',
+        fs_passno='0'
+    )
+]
+
+MOUNTS_INVALID = [
+    FstabEntry(
+        fs_spec='/dev/mapper/rhel_ibm--p8--kvm--03--guest--03-root',
+        fs_file='/',
+        fs_vfstype='xfs',
+        fs_mntops='defaults,nobarrier',
+        fs_freq='0',
+        fs_passno='0'
+    ),
+    FstabEntry(
+        fs_spec='UUID=c3ae2d1a-8088-45c2-9482-4aacb2c075cf',
+        fs_file='/boot',
+        fs_vfstype='xfs',
+        fs_mntops='defaults,barrier,noatime,delaylog,ihashsize',
+        fs_freq='0',
+        fs_passno='0'
+    )
+]
+
+MOUNTS_CORRECT_2 = [
+    FstabEntry(
+        fs_spec='/dev/mapper/luks-afe3592c-42c7-4702-bb1f-d403f63034c3',
+        fs_file='/',
+        fs_vfstype='ext4',
+        fs_mntops='defaults,x-systemd.device-timeout=0',
+        fs_freq='1',
+        fs_passno='1'
+    ),
+    FstabEntry(
+        fs_spec='UUID=df14b440-a7cd-4af0-aa4e-bfb6216813d9',
+        fs_file='/boot',
+        fs_vfstype='ext4',
+        fs_mntops='defaults',
+        fs_freq='1',
+        fs_passno='2'
+    )
+]
+
+
+def test_ok(current_actor_context):
+    fstab = MOUNTS_CORRECT_1 + MOUNTS_CORRECT_2
+    current_actor_context.feed(StorageInfo(fstab=fstab))
+    current_actor_context.run()
+    assert not current_actor_context.consume(Report)
+
+
+def test_removed_single(current_actor_context):
+    fstab = MOUNTS_CORRECT_1 + MOUNTS_INVALID[:1] + MOUNTS_CORRECT_2
+    current_actor_context.feed(StorageInfo(fstab=fstab))
+    current_actor_context.run()
+    assert current_actor_context.consume(Report)
+    for string in ['/dev/mapper/rhel_ibm--p8--kvm--03--guest--03-root', 'nobarrier']:
+        assert string in current_actor_context.consume(Report)[0].report['summary']
+
+
+def test_removed_multiple(current_actor_context):
+    fstab = MOUNTS_CORRECT_1 + MOUNTS_INVALID + MOUNTS_CORRECT_2
+    current_actor_context.feed(StorageInfo(fstab=fstab))
+    current_actor_context.run()
+    assert current_actor_context.consume(Report)
+    for string in ['/dev/mapper/rhel_ibm--p8--kvm--03--guest--03-root', 'nobarrier',
+                   'UUID=c3ae2d1a-8088-45c2-9482-4aacb2c075cf', 'barrier', 'delaylog', 'ihashsize']:
+        assert string in current_actor_context.consume(Report)[0].report['summary']
```

Comment:
Same for the tests. We may want to check if the inhibitor flag was raised.

Reply:
Alright.
Added severity and remediation too, somehow I missed those.